### PR TITLE
Milestone2: Add UI

### PR DESCRIPTION
**Pull request by odhiambo-ed**
In this milestone, I added the following:

    - Deleted all the boilerplate from CRA.
    - Created a directory called components.
    - Created a new Calculator.js file inside the Components
    - Developed just the UI of the calculator.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+/* eslint-disable no-eval */
+function Calculator() {
+  const [data, setData] = useState('');
+  const setValue = (e) => {
+    setData(data.concat(e.target.name));
+  };
+  const emptyValues = () => {
+    setData('');
+  };
+  const removeSomeValues = () => {
```
Comment (thecodechaser) [inline src/components/Calculator.js]: - [ ] Please use class-based component for the calculator as this is the major requirement of the project.

### See the image for reference 

image

File: src/components/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,473 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+/* eslint-disable no-eval */
+function Calculator() {
+  const [data, setData] = useState('');
```
Comment (thecodechaser) [inline src/components/Calculator.js]: - [ ] Your app is not rendering anything on the server, Please make sure your app runs fine and matches with the given design.

### See the image of the error.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @white3d👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import './App.css';
+
+import Calculator from './components/Calculator';
+
+function App() {
```
Comment (micheaol) [inline src/App.js]: As requested by the previous code reviewer, `All components should be class-based components` including your `app.js`

> * [ ]  Please use class-based component for the calculator as this is the major requirement of the project.
> 
> ### See the image for reference
> image

Kindly make the changes.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @white3d 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Github inline comments feature is used properly.
- You asked for the changes in a friendly and polite way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi  @white3d , 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/App.js (hunk 3 above)
Comment (odhiambo-ed) [reply src/App.js]: Hello @micheaol Thank you for the review. I have a concern here. App.ja is not part of the components. "Use class based components (All components should be class-based).
Once the Calculator is ready, import it and use it in your main component (App.js).
" Kindly read this statement again.

Comment (micheaol) [reply src/App.js]: Hi @white3d ,

Please note that `app.js` is part of your component, so it should be a class-based component, this is part of the project requirement. If you are getting an error message with `app.js` being a class-based component, please write your code like the code below: Just passing in an empty constructor and the error will go away.


```javascript
// ./src/components/Calculator.js

import React, { Component } from 'react';

class Calculator extends Component {

  constructor() {
    super();
    this.state = { };
  }
  render() {
    // reference as a js object
    return {
      <div>
       // Calculator app design goes here
      </div>
    };
  }
}

export default Calculator; // export Calculator component
```

I hope this helps.

**Comment by ShoiraTa** [comment]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @white3d , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**